arm-batch/2015-12-01/BatchManagement.json

```diff
@@ -0,0 +1,1340 @@
+          "format": "date-time",
+          "description": "Gets or sets the UTC time at which the storage URL will expire."
+        }
+      },
+      "description": "Response to an ApplicationOperations.AddApplicationPackage request."
+    },
+    "ApplicationPackage": {
+      "properties": {
+        "version": {
+          "type": "string",
+          "description": "Gets or sets the version of the application package. "
+        },
+        "state": {
+          "type": "string",
+          "description": "Gets or sets the current state of the application package.",
+          "enum": [
+            "pending",
+            "active",
+            "unmapped"
+          ],
+          "x-ms-enum": {
+            "name": "PackageState"
+          }
+        },
+        "format": {
+          "type": "string",
+          "description": "Gets or sets the format of the application package, if known."
+        },
+        "lastActivationTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the time at which the package was last activated, if the package is active."
+        }
+      },
+      "description": "Contains information about an application package."
+    },
+    "Application": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Gets or sets a string that uniquely identifies the application within the account."
+        },
+        "displayName": {
+          "type": "string",
+          "description": "Gets or sets the display name for the application."
+        },
+        "packages": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApplicationPackage"
+          },
+          "description": "Gets or sets the list of packages under this application."
+        },
+        "allowUpdates": {
+          "type": "boolean",
+          "description": "Gets or sets whether packages within the application may be overwritten using the same version string."
+        },
+        "defaultVersion": {
+          "type": "string",
+          "description": "Gets or sets which package to use if a client requests the application but does not specify a version."
+        }
+      },
+      "description": "Contains information about an application in a Batch account."
+    },
+    "GetApplicationPackageResult": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Gets or sets the id of the application."
+        },
+        "version": {
+          "type": "string",
+          "description": "Gets or sets the version of the application package. "
+        },
+        "state": {
+          "type": "string",
+          "description": "Gets or sets the current state of the application package.",
+          "enum": [
+            "pending",
+            "active",
+            "unmapped"
+          ],
+          "x-ms-enum": {
+            "name": "PackageState"
```

Comment:
I agree, it is not a required property. If not specified then the default is false. However, specifying it makes it explicit to the developers (us and partners) of what is happening. I was stressing this only for clarity and human readability.

Thanks,
Amar

From: Matthew Christopher [mailto:notifications@github.com]
Sent: Friday, March 11, 2016 3:59 PM
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Amar Zavery <amzavery@microsoft.com>
Subject: Re: [azure-rest-api-specs] Add ARM Batch swagger spec. (#172)


In arm-batch/2015-12-01/BatchManagement.json<https://github.com/Azure/azure-rest-api-specs/pull/172#discussion_r55906841>:

> +          "description": "Gets or sets the id of the application."

> +        },

> +        "version": {

> +          "type": "string",

> +          "description": "Gets or sets the version of the application package. "

> +        },

> +        "state": {

> +          "type": "string",

> +          "description": "Gets or sets the current state of the application package.",

> +          "enum": [

> +            "pending",

> +            "active",

> +            "unmapped"

> +          ],

> +          "x-ms-enum": {

> +            "name": "PackageState"

modelAsString isn't a required property -- why do we have to pick? If we should be specifying it, maybe it should be required by AutoRest?

—
Reply to this email directly or view it on GitHub<https://github.com/Azure/azure-rest-api-specs/pull/172/files#r55906841>.